elki/src/main/java/de/lmu/ifi/dbs/elki/math/statistics/dependence/MCDEDependenceMeasure.java

```diff
@@ -0,0 +1,173 @@
+     */
+    protected int m = 50;
+
+    /**
+     * Expected share of instances in slice (independent dimensions).
+     */
+    protected double alpha = 0.5;
+
+    /**
+     * Share of instances in marginal restriction (reference dimension).
+     * Note that in the original paper alpha = beta and as such there is no explicit distinction between the parameters.
+     */
+    protected double beta = 0.5;
+
+    /**
+     * Random generator.
+     */
+    protected RandomFactory rnd;
+
+    /**
+     * Constructor.
+     *
+     * @param m Monte-Carlo iterations
+     * @param alpha Expected share of instances in slice (under independence)
+     * @param beta Share of instances in marginal restriction (reference dimension)
+     * @param rnd Random source
+     */
+    public MCDEDependenceMeasure(int m, double alpha, double beta, RandomFactory rnd){
+        if((beta > 1.0) || beta <= 0.0) throw new AbortException("beta must be in ]0;1]");
+        if((alpha >= 1.0) || alpha <= 0.0) throw new AbortException("alpha must be in ]0;1[ ");
+        if(m < 1) throw new AbortException("m must be > 0");
+
+        this.m = m;
+        this.alpha = alpha;
+        this.beta = beta;
+        this.rnd = rnd;
+    }
+
+    /**
+     * Overloaded wrapper for corrected_ranks()
+     */
+    protected <A> R[] corrected_ranks(final NumberArrayAdapter<?, A> adapter, final A data, int len) {
+        return corrected_ranks(adapter, data, sortedIndex(adapter, data, len));
+    }
+
+    /**
+     * Subclass must implement computation of corrected rank index.
+     *
+     * @param adapter ELKI NumberArrayAdapter Subclass
+     * @param data One dimensional array containing one dimension of the data
+     * @param idx Return value of sortedIndex()
+     * @return Array of RankStruct, acting as rank index
+     */
+    protected abstract <A> R[] corrected_ranks(final NumberArrayAdapter<?, A> adapter, final A data, int[] idx);
+
+    /**
+     * Subclass must implement the computation of the statistical test, based on the slicing scheme.
+     *
+     * @param len No of data instances
+     * @param slice An array of boolean resulting from a random slice
+     * @param corrected_ranks the precomputed index structure for the reference dimension
+     * @return a 1 - p-value
+     */
+    protected abstract double statistical_test(int len, boolean[] slice, R[] corrected_ranks);
+
+    /**
+     * Data Slicing
+     *
+     * @param len No of data instances
+     * @param nonRefIndex Index (see correctedRank()) computed for the dimension that is not the reference dimension
+     * @return Array of booleans that states which instances are part of the slice
+     */
+    protected boolean[] randomSlice(int len, R[] nonRefIndex){
+        final Random random = rnd.getSingleThreadedRandom();
+        boolean slice[] = new boolean[len];
+        Arrays.fill(slice, Boolean.TRUE);
+
+        final int slizeSize = (int) Math.ceil(Math.pow(this.alpha, 1.0) * len);
```

Comment:
`pow(x,1)` clearly is an error, maybe a leftover from an earlier version that used a different exponent?